src/wp-includes/blocks.php

```diff
@@ -2420,6 +2421,52 @@ function build_query_vars_from_query_block( $block, $page ) {
 				}
 			}
 		}
+		if ( ! empty( $block->context['query']['format'] ) && is_array( $block->context['query']['format'] ) ) {
+			$formats   = $block->context['query']['format'];
+			$tax_query = array( 'relation' => 'OR' );
```

Comment:
My reading of this code is that if someone sets up a query for both formats and taxomomies, this will return posts that contain either.

Eg a query for `post format: link; category: rest-api` will return:
* Any posts with the post format link, regardless of category
* Any posts in the category rest-api, regardless of format.

Reply:
Yeah I think something is not working correctly.
If I use `OR` or `AND`, the result is the same: Posts that have **both** the category and the format. 
I think the correct use is `AND`.
But then why is not `OR` returning all posts that have either the format or the category?

Reply:
Here is an example query loop for listing posts in category 4, with either the gallery or link format. The category id needs to be updated to match category available on the installation.

```
<!-- wp:query {"queryId":1,"query":{"perPage":10,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false,"format":["link","gallery"],"taxQuery":{"category":[4]}}} -->
<div class="wp-block-query"><!-- wp:post-template -->
<!-- wp:post-title /-->
<!-- /wp:post-template --></div>
<!-- /wp:query -->
```

Reply:
I have changed it to `AND` though it bothers me why `OR` is working the way I am seeing it in testing 😄

Reply:
🤔 In the `WP_REST_Posts_Controller` class, there is also:

```
// Enable filtering by both post formats and other taxonomies by combining them with `AND`.
			if ( isset( $args['tax_query'] ) ) {
				$args['tax_query'][] = array(
					'relation' => 'AND',
					$tax_query,
				);
			} else {
				$args['tax_query'] = $tax_query;
			}
```
https://github.com/WordPress/wordpress-develop/pull/7314/files#diff-48cfdad3b747164ac4b6857186399519dabf913026235575b00ebe5b0dc7169bR383

Reply:
I am badly regretting not documenting why I add this relation in the first place.

Once I get back to work I will continue doing more logging of the different queries:
But I am now thinking that I meant for this to be the relation **between the formats**
That I choose 'OR' because the user can query for both 'video' and 'link' and it needs to be 'OR' because a post can only have one format. It always have to list posts that has either 'video' or 'link', never both.

Reply:
Also I think `$tax_query` was not a good name for the variable, it too can cause confusion.

Reply:
After more testing and logging of `$args['tax_query']` in the `WP_REST_Posts_Controller` class, I found that:

- Setting the relation on the array to either `OR` or `AND` works when the query has two formats such as link and gallery. Because there is only one condition; the relation is not actually used, there are no conditions to create a relation between.

- Using `OR` is needed when the query includes `standard` **and** a non-default format such as link and gallery, because now there are two conditions.

So that is why the testing earlier showed the same results: I was not testing with a combination of `standard`.